server/src/main/java/org/elasticsearch/action/ActionListener.java

```diff
@@ -62,7 +62,15 @@ protected Delegating(ActionListener<DelegateResponse> delegate) {
 
         @Override
         public void onFailure(Exception e) {
-            delegate.onFailure(e);
+            try {
+                delegate.onFailure(e);
```

Comment:
I think it would be nice to make `delegate` private to ensure that nobody does `delegate.onFailure` in the future now that you cleared out all of them.

I think adding a protected `delegateOnResponse` method to actually delegate the call would be possible though i did not check all usages.

Reply:
Making `delegate` private is a little tricky (at least for now). We use that field in various spots and at least in `DelegatingActionListener` a few lines down it's not straight forward to make it private without further changes.

I'd rather push making this safe against extensions to a follow-up I think.

Reply:
I would think it could work fine with package-private. But of course also OK to defer to a follow-up.